.github/workflows/tests.yml

```diff
@@ -0,0 +1,43 @@
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Setup Micromamba
+      uses: mamba-org/provision-with-micromamba@v13
+      with:
+        environment-file: false
+
+    - name: Python ${{ matrix.python-version }}
+      shell: bash -l {0}
+      run: |
+        micromamba create --yes --name TEST python=${{ matrix.python-version }} pip python-build --file requirements.txt --file requirements-dev.txt --channel conda-forge
+        micromamba activate TEST
+        pip install -e . --no-deps --force-reinstall
+    - name: Tarball tests
+      shell: bash -l {0}
+      run: |
+        micromamba activate TEST
+        python setup.py --version
+        python -m build --sdist --wheel . --outdir dist
+        check-manifest --verbose
+        twine check dist/*
```

Comment:
I'm leaning towards separating the tarball tests into its own action named pypi.yml, that tests and publishes the sdist and wheel. Let's remove this for now and we can work on that in another PR.

```suggestion
```